core/remotes/docker/fetcher.go

```diff
@@ -262,7 +263,7 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest, op
 	return seeker, desc, nil
 }
 
-func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
+func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64, remainingHosts bool) (_ io.ReadCloser, retErr error) {
```

Comment:
nit: I feel `lastHost` is easier to understand than `remainingHosts`.

with `remainingHosts` we're saying "retry if not remainingHosts"
with `lastHost` we're saying "retry if lastHost"